.github/workflows/pr-build.yml

```diff
@@ -37,9 +66,41 @@ jobs:
       - name: Build with Flutter
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        run: flutter build apk --debug
+        run: |
+          flutter build apk --${{ inputs.app-variant }};
+      - name: Set env
+        run: |
+          echo "COMMIT_HASH=$(git rev-parse --short HEAD)" >> $GITHUB_ENV
       - name: Upload build
         uses: actions/upload-artifact@v3
         with:
-          name: revanced-manager
-          path: build/app/outputs/flutter-apk/app-debug.apk
+          name: revanced-manager-${{ env.COMMIT_HASH }}-${{ inputs.pr-number }}-${{ inputs.app-variant }}-${{ inputs.flutter-variant }}
+          path: build/app/outputs/flutter-apk/app-${{ inputs.app-variant }}.apk
+          
+  notify-pull-request:
+    name: Notify Pull Request
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - name: Set env
+        run: |
+          echo "DATETIME=$(date +'%Y-%m-%d %H:%M:%S')" >> $GITHUB_ENV
+          if [[ "${{ needs.build.result }}" == "failure" ]]; then
+            echo "MESSAGE=🚫 Build failed!" >> $GITHUB_ENV
+          else
+            echo "MESSAGE=✅ Build succeeded!" >> $GITHUB_ENV
+          fi
+      - name: "Comment to Pull Request #${{ inputs.pr-number }}"
+        uses: thollander/actions-comment-pull-request@v2
+        with:
+          GITHUB_TOKEN: ${{ github.token }}
+          pr_number: ${{ inputs.pr-number }}
+          mode: recreate
+          message: |
+            ${{ env.MESSAGE }} @ ${{ env.DATETIME }}
+            
+            You can check the details [here](${{ github.event.workflow_run.html_url }}).
```

Comment:
TODO: Fix URL... ????? HOW

Why is this linked to the PR's URL, not the workflow url

Reply:
Where did you see that it's returning the PR's URL?

Do you mind sharing the repository where you're testing this action?

Reply:
https://github.com/validcube/revanced-manager/pull/17#issuecomment-1739217455